src/styles/annotator/components/Buckets.scss

```diff
@@ -1,48 +1,107 @@
-@use '../mixins/buttons';
-@use '../variables' as var;
-
-$bucket-bar-width: 22px;
-
-.Buckets__list {
-  background: var.$grey-2; // When sidebar is unfolded, remove the background transparency
-  pointer-events: none;
-  position: absolute;
-  height: 100%;
-  // 2020-11-20: interim and pragmatic solution for an apparent glitch on Safari and Chrome.
-  // Adding one pixel resolve this issue: https://github.com/hypothesis/client/pull/2750
-  width: $bucket-bar-width + 1;
-  left: -($bucket-bar-width);
-}
+/**
+ * These classes style buttons to appear as lozenges with an integrated arrow
+ * pointing left, up or down (right doesn't exist yet but could easily be added).
+ * *
+ * The arrow-points are created by the combination of borders and positioning.
+ * See https://css-tricks.com/snippets/css/css-triangle/ for a few examples
+ *
+ */
 
-// When the sidebar is collapsed, make the background semi-transparent so the
-// text is visible throughout (useful for pages with tight margins)
-.annotator-collapsed .Buckets__list {
-  background: rgba(0, 0, 0, 0.08);
-}
+@layer components {
+  // Base styling for a button with a numerical badge that points either up,
+  // down or left. On its own, it will be lozenge-shaped. Compose with one
+  // of the directional classes below to style a button fully, e.g.
+  // `classname="BucketButton LeftBucketButton`
+  .BucketButton {
+    // A lozenge-shaped element with very small text
+    @apply w-[26px] h-[16px] absolute border bg-white shadow rounded;
+    @apply font-sans text-center text-tiny font-bold text-color-text-light leading-none;
 
-.Buckets__bucket {
-  position: absolute;
-  right: 0;
-}
+    // Establish :before and :after content for later manipulation into
+    // different pointer shapes and directions
+    &::before,
+    &::after {
+      @apply content-[""] absolute border-transparent;
+      // NB: use of 'inset' here fixes jagged diagonals in FF
+      // https://github.com/zurb/foundation/issues/2230
```

Comment:
This issue links to a Firefox bug report at https://bugzilla.mozilla.org/show_bug.cgi?id=804747 which is still open. This in turn links to another bug report which has been closed as "seems to be resolved" - https://bugzilla.mozilla.org/show_bug.cgi?id=804747.

In my testing the rendering seemed correct without the `border: inset` rule. This rule also caused a rendering problem in Safari where the border color appeared lighter than expected.

Reply:
Yeah, I actually did click through when I set out on this work, but wasn't convinced by what I found (the whole "seems to be resolved" thing). At the exact time I was looking into that, I was having trouble getting the buttons to render correctly in FireFox—almost 100% assuredly for other reasons. I'm going to look into this again, because the lack of support for this in Tailwind suggests that this is a pretty obscure workaround.